Extension/src/LanguageServer/extension.ts

```diff
@@ -104,6 +120,9 @@ function realActivation(): void {
 
     reportMacCrashes();
 
+    eventEmitter.emit(activationCompleteEventId);
```

Comment:
Having this event may be unnecessary if we just keep track of the providers in this file and send them down to the Client when they are available.  I haven't thought this all the way through yet.  I might be able to be more prescriptive once you get the UI part in for selecting providers.